src/util/sample.cpp

```diff
@@ -660,11 +860,44 @@ void SampleUtil::reverse(CSAMPLE* pBuffer, SINT numSamples) {
 }
 
 // static
-void SampleUtil::copyReverse(CSAMPLE* M_RESTRICT pDest,
-        const CSAMPLE* M_RESTRICT pSrc, SINT numSamples) {
+void SampleUtil::copyReverseStereo(CSAMPLE* M_RESTRICT pDest,
+        const CSAMPLE* M_RESTRICT pSrc,
+        SINT numSamples) {
     for (SINT j = 0; j < numSamples / 2; ++j) {
```

Comment:
I wonder if it's neccessary to calculate endpos every iteration again. Wouldn't something like 
```
// static
void SampleUtil::copyReverseStereo(CSAMPLE* M_RESTRICT pDest,
        const CSAMPLE* M_RESTRICT pSrc,
        SINT numSamples) {
    pSrc += numSamples - 2;
    for (SINT j = 0; j < numSamples; j += 2) {
        pDest[j] = *pSrc--;
        pDest[j + 1] = *pSrc--;
    }
}
```
do the same?

Reply:
The loop unrolling by the fixed values of 2 or 8 helps the compiler with vectorization, because odd channel numbers do not need to be supported and it can pick the optimal register size. Don't know if this constraints are enough for vectorization, but I think it will help.